runtime/src/common/sgx/seal.rs

```diff
@@ -49,11 +53,11 @@ pub fn unseal(key_policy: Keypolicy, context: &[u8], ciphertext: &[u8]) -> Optio
     let ciphertext = &ciphertext[..ct_len];
 
     let d2 = new_deoxysii(key_policy, context);
-    let plaintext = d2
-        .open(&nonce, ciphertext.to_vec(), vec![])
-        .expect("ciphertext is corrupted");
 
-    Some(plaintext)
+    match d2.open(&nonce, ciphertext.to_vec(), vec![]) {
+        Ok(plaintext) => Ok(Some(plaintext)),
+        Err(e) => Err(format_err!("ciphertext is corrupted (if running in SGX mode, check if the CPU had changed; if yes, wipe 'worker-local-storage.badger.db' and resync the node): {}", e.to_string())),
```

Comment:
I would not reveal to many information why `open` failed.
```suggestion
        Err(e) => Err(format_err!("ciphertext is corrupted (if running in SGX mode, check if the CPU had changed; if yes, wipe 'worker-local-storage.badger.db' and resync the node)")),
```